tools/ldb_cmd_test.cc

```diff
@@ -282,6 +282,10 @@ TEST_F(LdbCmdTest, DumpFileChecksumNoChecksum) {
   Options opts;
   opts.env = env.get();
   opts.create_if_missing = true;
+  // To avoid WAL sync during shutdown as writable file in mock env is
+  // not sync thread safe so we can't sync WAL and DB Close()
+  // will return non-OK
+  opts.avoid_flush_during_shutdown = true;
```

Comment:
This test reminded me of something: db close() can now return non-okay if SyncWAL() fails or not supported if writable file implementation does not support "sync thread safe" https://github.com/facebook/rocksdb/blob/1fca175eece9213a07f99973bae8e9a7d6aea93c/db/db_impl/db_impl.cc#L1572

This means users can see new non-okay error during DB close. Does this behavior change concern you all? @ajkr  @ltamasi

Reply:
Joining the party little late. To me, non-OK status for close() is okay as long as we clearly tell users that they need to retry the close(). e.g. We already return `Status::Aborted()` when there are snapshots that cannot be released during `close()` today.